apps/ui/components/common/Navigation.tsx

```diff
@@ -109,29 +126,52 @@ function constructName(firstName?: string, lastName?: string) {
 }
 
 function NavigationMenu({ user }: { user: CurrentUserQuery["currentUser"] }) {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const { pathname } = useLocation();
+  const router = useRouter();
+
+  const handleClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+    router.push(e.currentTarget.href);
+  };
+
   return (
     <Header.NavigationMenu>
       {menuItems.map((item) => {
         if (item.requireLogin && !user) {
           return null;
         }
+        const localisationString =
+          i18n.language === "fi" ? "" : getLocalizationLang(i18n.language);
         const isActive = () => {
           if (!pathname) return false;
           let active = false;
-          item.href.forEach((href) => {
-            if (pathname.startsWith(href)) {
+          item.routes.forEach((route) => {
+            const localizedPath =
+              localisationString !== ""
+                ? `/${localisationString}${route}`
+                : route;
+            if (
+              item.exact
+                ? pathname === localizedPath
+                : pathname.startsWith(localizedPath)
+            ) {
               active = true;
             }
           });
           return active;
         };
+
         return (
-          <Header.Link
+          <Header.ActionBarSubItem
             key={item.label}
             label={t(item.label)}
-            href={item.href[0]}
+            href={
+              localisationString === ""
+                ? item.routes[0]
+                : `/${localisationString}${item.routes[0]}`
+            }
```

Comment:
check against lang: "fi" | "sv" | "en", not a tmp string that is modified

Reply:
now using getLocalizationLanguage() instead